internal/acceptance/entitlements_test.go

```diff
@@ -95,3 +103,122 @@ func TestAccServicePrincipalEntitlementsResourceOnAws(t *testing.T) {
 		}`,
 	})
 }
+
+type entitlement struct {
+	name  string
+	value bool
+}
+
+func (e entitlement) String() string {
+	return fmt.Sprintf("%s = %t", e.name, e.value)
+}
+
+func entitlementsStepBuilder(t *testing.T, c **httpclient.ApiClient, groupName string) func(entitlements []entitlement) step {
+	return func(entitlements []entitlement) step {
+		entitlementsBuf := strings.Builder{}
+		for _, entitlement := range entitlements {
+			entitlementsBuf.WriteString(fmt.Sprintf("%s\n", entitlement.String()))
+		}
+		return step{
+			Template: fmt.Sprintf(`
+			data "databricks_group" "example" {
+				display_name = "%s"
+			}
+			
+			resource "databricks_entitlements" "entitlements_users" {
+				group_id              = data.databricks_group.example.id
+				%s
+			}
+		`, groupName, entitlementsBuf.String()),
+			Check: func(s *terraform.State) error {
+				groupId := s.RootModule().Resources["data.databricks_group.example"].Primary.ID
+				var res iam.Group
+				ctx := context.Background()
+				err := (*c).Do(ctx, "GET", fmt.Sprintf("/api/2.0/preview/scim/v2/Groups/%s?attributes=entitlements", groupId),
+					httpclient.WithResponseUnmarshal(&res))
+				assert.NoError(t, err)
+				receivedEntitlements := make([]string, 0, len(res.Entitlements))
+				for _, entitlement := range res.Entitlements {
+					receivedEntitlements = append(receivedEntitlements, entitlement.Value)
+				}
+				expectedEntitlements := make([]string, 0, len(entitlements))
+				for _, entitlement := range entitlements {
+					if entitlement.value {
+						expectedEntitlements = append(expectedEntitlements, strings.ReplaceAll(entitlement.name, "_", "-"))
+					}
+				}
+				assert.ElementsMatch(t, expectedEntitlements, receivedEntitlements)
+				return nil
+			},
+		}
+
+	}
+}
+
+func makeEntitlementsSteps(t *testing.T, entitlementsSteps [][]entitlement) []step {
+	groupName := RandomName("entitlements-")
+	var c *httpclient.ApiClient
+	makeEntitlementsStep := entitlementsStepBuilder(t, &c, groupName)
+	steps := make([]step, len(entitlementsSteps))
+	for i, entitlements := range entitlementsSteps {
+		steps[i] = makeEntitlementsStep(entitlements)
+	}
+	steps[0].PreConfig = makePreconfig(t, &c, groupName)
+	return steps
+}
+
+func makePreconfig(t *testing.T, c **httpclient.ApiClient, groupName string) func() {
+	return func() {
+		w := databricks.Must(databricks.NewWorkspaceClient())
+		var err error
+		*c, err = w.Config.NewApiClient()
+		assert.NoError(t, err)
+		ctx := context.Background()
+		group, err := w.Groups.Create(ctx, iam.Group{
+			DisplayName: groupName,
+		})
+		assert.NoError(t, err)
+		t.Cleanup(func() {
+			err := w.Groups.DeleteById(ctx, group.Id)
+			assert.NoError(t, err)
+		})
+	}
+}
+
+func TestAccEntitlementsAddToEmpty(t *testing.T) {
+	steps := makeEntitlementsSteps(t, [][]entitlement{
+		{},
+		{
+			{"allow_cluster_create", true},
+			{"allow_instance_pool_create", true},
+			{"workspace_access", true},
+			{"databricks_sql_access", true},
```

Comment:
```suggestion
			{"allow_cluster_create", false},
			{"allow_instance_pool_create", false},
			{"workspace_access", false},
			{"databricks_sql_access", false},
```

Reply:
I'll make a separate test for this, as I did want to have one test that sets all entitlements to true on create.